### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by CalvinScott**
Add some basic tests to this repo. There isn't much here, but we should at least insure that there aren't any syntax errors in the JS.

### CODE REVIEW

**Comment by CalvinScott** [comment]
@CMClay @RongxinZhang Ready to review, let me know if there's anything I can change/add. 

I realize I probably did a bit too much (strenuous syntax tests) for some of the files that I started with, but Rongxin tells me that there might be some space for these in some api test files.

**Comment by RongxinZhang** [comment]
@CalvinScott
- need to add mocha script to package.json
- add ```npm test``` towercker.yml
- The tests themselves need to be checked with @CMClay.

**Comment by CalvinScott** [comment]
Definitely good practice in JS syntax, looks like my go-to looping structure is out the window...probably ready to review now. @CMClay @RongxinZhang

File: test/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+test('Check Schemas', function(){
+	for(var i = 0; i < schemas.length; i++){	// beware for-loops that have function callbacks inside!
+ 		assert('object' === typeof schemas[i], 'object exists')
+	}
+});
+
+test('Parse Contents', function(){
+	schemas.forEach(function(schema){		// traverses each element in the schemas array
+		Object.keys(schema).forEach(function(key){		// creates an array of all the enumurable keys for each schema, and traverses them
+			assert(schema[key].indexOf != -1, 'content exists')		// checks each element to see if at has a correctly associated key
```
Comment (CMClay) [inline test/index.js]: always use strict comparison: `===` or `!==`

**Comment by CMClay** [comment]
@CalvinScott I thought there was going to be a test that all the `type` fields are of a valid type. For example:
```js
var types = ['string', 'date', 'object', 'array'];
assert(types.indexOf(schema[key].type) !== -1);
```

also can test that the `subtype` field is `typeof === 'object'`

File: test/index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+var schemas = [index, agent, media, office, geo, listing, openhouse];
+
+/**
+ * Tests
+ */
+
+suite('Object Creation')
+test('Check Schemas', function(){
+	for(var i = 0; i < schemas.length; i++){	// beware for-loops that have function callbacks inside!
+ 		assert('object' === typeof schemas[i], 'object exists')
```
Comment (CMClay) [inline test/index.js]: semicolon

File: test/index.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+var listing = require('../lib/listing.js');
+var openhouse = require('../lib/openhouse.js');
+
+var schemas = [index, agent, media, office, geo, listing, openhouse];
+
+/**
+ * Tests
+ */
+
+suite('Object Creation')
```
Comment (CMClay) [inline test/index.js]: semicolon

**Comment by CalvinScott** [comment]
Spent about an hour trying to figure out why sometimes traversing with Object.key...forEach() and then using .type would give undefined. Turns out it had something to do with depth - since index has all of the other schemas within it, it has to go an extra layer to get to the type, if that makes sense. By using another array without index solved the problem (whereas an extra Object.keys...forEach loop had other issues).

On another note I am still trying to figure out how to check the subtype field.

**Comment by CMClay** [comment]
@CalvinScott I think your indentation settings in sublime text are messed up.
You have to make sure your tabs are translated to spaces.
Add this to **Preferences** => **Settings-User**
	"tab_size": 2,
	"translate_tabs_to_spaces": true

File: test/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+suite('Object Creation');
+test('Check Schemas', function(){
+	for(var i = 0; i < schemas.length; i++){
+ 		assert('object' === typeof schemas[i], 'object exists');
+	}
+});
+
+test('Parse Contents', function(){
+	subschemas.forEach(function(schema){
+		//console.log('first',schema);
```
Comment (CMClay) [inline test/index.js]: You can take these out.

File: test/index.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+		Object.keys(schema).forEach(function(key){	
+			//console.log('second',schema[key]);
+			assert(types.indexOf(schema[key].type) !== -1, 'content exists');
+			if(typeof schema[key].subtype !== 'undefined'){
```
Comment (CMClay) [inline test/index.js]: Simply check like this `if (typeof schema[key].subtype)`

File: test/index.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+var media = require('../lib/media.js');
+var office = require('../lib/office.js');
+var geo = require('../lib/geo.js');
+var listing = require('../lib/listing.js');
+var openhouse = require('../lib/openhouse.js');
+
+var schemas = [index, agent, media, office, geo, listing, openhouse];
+var subschemas = [agent, media, office, geo, listing, openhouse]; 
+var types = ['object', 'string', 'date', 'array', 'number', 'boolean'];
+var subtypes = ['object', 'string', 'number'];
```
Comment (CMClay) [inline test/index.js]: Subtype can be a number?

**Comment by CMClay** [comment]
@CalvinScott Almost there!

**Comment by CalvinScott** [comment]
@CMClay @slajax 
Hopefully ready to merge now!

File: test/index.js (hunk 6 above)
Comment (slajax) [inline test/index.js]: @CalvinScott, I think @CMClay is right.

**Comment by slajax** [comment]
:+1:

**Comment by CMClay** [comment]
:+1:

**Pull request merged**